draft-ietf-ppm-dap.md

```diff
@@ -706,6 +706,176 @@ last two weeks of the current time) and reject reports that fall outside of the
+      batch bucket 1  batch bucket 1   bucket 2        bucket 2    .-----'
+             |            |                |               |       |
+             |<-----------|----------------'               |       v
+             |            |<-------------------------------' .-----+-----.
+                                                             |           |
+            ...          ...                          Leader batch  Helper batch
+                                                        bucket j      bucket j
+             |            |                                  |           |
+             |<-----------|----------------------------------'           |
+             |            |<---------------------------------------------'
+             |<-----------+<----- query chosen
+             |            |       by Collector
+         .---+------------+---.
+        | batch buckets merged |
+         '---+------------+---'
+             |            |
+             v            v
+   Leader aggregate  Helper aggregate
+          share         share
+             |            |
+             '------+-----'
+                    |
+               .----+----.
+              | unsharded |
+               '----+----'
+                    |
+                    v
+             aggregate result
+~~~
+{: #object-lifecycle title="Lifecycles of protocol objects" }
+
+### Arity of Protocol Objects
+
+DAP reports are 1:1 to with measurements. In this illustration, `i` distinct
+Clients upload a distinct report, but a single Client could upload multiple
+reports to a task (see {{sybil}} for some implications of this). The process of
+sharding measurements, constructing reports and uploading them is discussed in
+detail in {{upload-flow}}.
+
+Reports are many:1 with aggregation jobs. The Leader assigns each of the `i`
+reports into one of `j` different aggregation jobs, which can be run in parallel
+by the Aggregators. Each aggregation job contains `k` report shares. `k` is
+roughly `i / j`, but it is not necessary for aggregation jobs to have uniform
+size. See {{operational-capabilities}} for some discussion of performance
+implications for aggregation job scheduling strategies.
+
+Report shares, input shares and output shares have a 1:1:1 relationship. Report
+shares are decrypted into input shares and then prepared into output shares. The
+scheduling of aggregation jobs and their execution by the Aggregators is
+discussed in detail in {{aggregate-flow}}.
+
+Output shares are accumulated into batch buckets, and so have a many:1
+relationship. In this example, we show one batch bucket for each aggregation
+job, but aggregation jobs and batch buckets are not necessarily 1:1. Multiple
+aggregation jobs could contribute to the same batch bucket, and a single
+aggregation job could contribute to multiple batch buckets. The assignation of
+output shares to batch buckets is an implementation detail of each Aggregator.
```

Comment:
I'm wondering to what extent this is true. The batch mode defines an infinite sequence of non-overlapping batch buckets. I suppose the aggregators could sub-divide these further if they wanted to. Is that what you mean?

Reply:
I'm not following. My understanding of "batch bucket" as we define it in the spec:

1. time_interval: [0, t), [t, 2*t), [2*t, 3*t), ..., where t is short for time_precision
2. leader_selected: batch_id.

At aggregation time, the aggregator always knows which batch bucket a report belongs to.

Reply:
You are correct. I hadn't noticed we'd defined batch buckets that rigorously in [4.8.2.3](https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-14#section-4.8.2.3), [5.1.4](https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-14#section-5.1.4) and [5.2.4](https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-14#section-5.2.4). Thanks for pointing that out. I fixed this text to refer to `{{batch-buckets}}`.